src/main/java/org/opensearch/geospatial/index/common/shape/ShapeConverter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.common.shape;
+
+import static org.apache.commons.lang3.ArrayUtils.toPrimitive;
+
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+
+import org.apache.lucene.geo.XYLine;
+import org.apache.lucene.geo.XYPolygon;
+import org.opensearch.geometry.Line;
+import org.opensearch.geometry.Polygon;
+import org.opensearch.geometry.Rectangle;
+
+public class ShapeConverter {
+
+    public static XYLine toXYLine(Line line) {
+        float[] x = toFloatArray(line.getX());
+        float[] y = toFloatArray(line.getY());
+        return new XYLine(x, y);
+    }
+
+    private static float[] toFloatArray(double[] input) {
+        final Float[] floats = DoubleStream.of(input).boxed().map(Double::floatValue).toArray(Float[]::new);
+        return toPrimitive(floats);
+    }
+
+    public static XYPolygon toXYPolygon(Rectangle r) {
+        // build polygon by assigning points in Counter Clock Wise direction (default for polygon) and end at where
+        // you started since it has to be linear ring
+        // (minX,minY) -> (maxX+minY) -> (maxX, maxY) -> (minX, maxY) -> (minX,minY)
+        double[] x = new double[] { r.getMinX(), r.getMaxX(), r.getMaxX(), r.getMinX(), r.getMinX() };
+        double[] y = new double[] { r.getMinY(), r.getMinY(), r.getMaxY(), r.getMaxY(), r.getMinY() };
+        return new XYPolygon(toFloatArray(x), toFloatArray(y));
+    }
+
+    public static XYPolygon toXYPolygon(Polygon polygon) {
```

Comment:
How we are validating that the arguments passed over here and at all other places in the public function are not null?

Reply:
Initially i was validating inside the indexer. Now moved the validation here.